source/code/plugins/oms_common.rb

```diff
@@ -478,6 +482,118 @@ class Common
     @@tzRightFolder = 'right/'
 
     class << self
+
+      # Internal methods
+      # (left public for easy testing, though protected may be better later)
+
+      def clean_hostname_string(hnBuffer)
+        hostname_buffer = hnBuffer.strip
+        return hostname_buffer
+      end
+
+      def has_designated_hostnamefile?
+        return false if @@HostnameFilePath.nil?
+        return false unless @@HostnameFilePath =~ /\w/
+        return false unless File.exist?(@@HostnameFilePath)
+        return true
+      end
+
+      def is_dot_separated_string?(hnBuffer)
+        return true if /[^.]+\.[^.]+/ =~ hnBuffer
+        return false
+      end
+
+      def is_hostname_compliant?(hnBuffer)
+        # RFC 2181:
+        #   Size limit is 1 to 63 octets, so probably bytesize is appropriate method.
+        return false if hnBuffer.nil?
+        return false if /\./ =~ hnBuffer # Hostname by definition may not contain a dot.
+        return false if /:/ =~ hnBuffer # Hostname by definition may not contain a colon.
+        return false unless 1 <= hnBuffer.bytesize && hnBuffer.bytesize <= 63
+        return true
+      end
+
+      def is_like_ipv4_string?(hnBuffer)
+        return false unless /\A#{IPV4_Approximate_REGEX}\z/ =~ hnBuffer
+        qwa = hnBuffer.split('.')
+        return false unless qwa.length == 4
+        return false unless qwa[0].to_i != 0
```

Comment:
This check has an unnecessary double-negative: to me, it reads easier as `return false if qwa[0].to_i == 0`